SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/CrossServerAbilityActivationTest/CrossServerAbilityActivationTest.cpp

```diff
@@ -0,0 +1,186 @@
+ * - Cleanup:
+ *	- The test actor is registered to be auto-destroyed by the test framework at test end.
+ */
+
+ACrossServerAbilityActivationTest::ACrossServerAbilityActivationTest()
+{
+	Author = "Tilman Schmidt";
+	Description = TEXT("Tests that gameplay abilities can be activated across server boundaries.");
+	DuplicateActivationCheckWaitTime = 2.0f;
+
+	TargetActor = nullptr;
+	StepTimer = -1.0f;
+}
+
+bool TargetActorReadyForActivation(ASpyValueGASTestActor* Target)
+{
+	return Target->GetLocalRole() == ENetRole::ROLE_SimulatedProxy && Target->GetCounter() == 0;
+}
+
+void ACrossServerAbilityActivationTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	FName TargetActorTag = FName(TEXT("Target"));
+
+	// Step 1 - Spawn Actor On Auth
+	AddStep(TEXT("Spawn actor"), FWorkerDefinition::Server(1), nullptr, [this, TargetActorTag]() {
+		UWorld* World = GetWorld();
+		TargetActor = World->SpawnActor<ASpyValueGASTestActor>({ 0.0f, 0.0f, 0.0f }, FRotator::ZeroRotator);
+		AssertTrue(IsValid(TargetActor), TEXT("Target actor is valid."));
+		AssertTrue(TargetActor->GetCounter() == 0, TEXT("Target actor spy counter starts at 0"));
+		AddDebugTag(TargetActor, TargetActorTag);
+		RegisterAutoDestroyActor(TargetActor);
+		FinishStep();
+	});
+
+	AddStepSetTagDelegation(TargetActorTag, 2);
+
+	// Step 2 - Trigger an ability by spec handle, cross-server, and confirm that it got activated
+	AddStep(
+		TEXT("Activate Ability by Spec Handle"), FWorkerDefinition::Server(1),
+		[this]() {
+			return TargetActorReadyForActivation(TargetActor);
+		},
+		[this]() {
+			UAbilitySystemComponent* ASC = TargetActor->GetAbilitySystemComponent();
+			AssertIsValid(ASC, TEXT("TargetActor has an ability system component."));
+
+			TArray<FGameplayAbilitySpec*> FoundAbilitySpecs;
+			ASC->GetActivatableGameplayAbilitySpecsByAllMatchingTags(FGameplayTagContainer(UGA_IncrementSpyValue::GetTriggerTag()),
+																	 FoundAbilitySpecs);
+			AssertEqual_Int(FoundAbilitySpecs.Num(), 1, TEXT("Target actor has one granted instance of the ability"));
+			FGameplayAbilitySpecHandle AbilityHandle = FoundAbilitySpecs[0]->Handle;
+			AssertTrue(AbilityHandle.IsValid(), TEXT("Activatable ability has a valid handle"));
+
+			ASC->CrossServerTryActivateAbility(AbilityHandle);
+		},
+		[this](float DeltaTime) {
+			bool bShouldFinishStep = WaitForActivationConfirmation(DeltaTime);
+			if (bShouldFinishStep)
+			{
+				FinishStep();
+			}
+		});
+
+	// Step 3 - Same as 2, but activate by class
+	AddStep(
+		TEXT("Activate Ability by Class"), FWorkerDefinition::Server(1),
+		[this]() {
+			return TargetActorReadyForActivation(TargetActor);
+		},
+		[this]() {
+			UAbilitySystemComponent* ASC = TargetActor->GetAbilitySystemComponent();
+			AssertIsValid(ASC, TEXT("TargetActor has an ability system component."));
+
+			ASC->CrossServerTryActivateAbilityByClass(UGA_IncrementSpyValue::StaticClass());
+		},
+		[this](float DeltaTime) {
```

Comment:
I assume we can reuse the same lambda for every TickEvent rather than write it in every AddStep?
```
auto CheckAbilityConfirmed = [this](float DeltaTime) {
			bool bShouldFinishStep = WaitForActivationConfirmation(DeltaTime);
			if (bShouldFinishStep)
			{
				FinishStep();
			}
		};

AddStep(..., CheckAbilityConfirmed);
```